compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -856,6 +856,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             format!("impl {}", all_bounds_str),
             Applicability::MaybeIncorrect,
         );
+        err.note(format!("the type of `{}` is chosen by the caller", expected_ty_as_param.name));
```

Comment:
I'm not sure if this note adds much value in this context. Not exactly sure why, but something about the ordering or placement of this note feels to me like it's missing context.

In the `` stderr below:

```
LL | fn used_in_trait<T>() -> T
   |                  -       -
   |                  |       |
   |                  |       expected `T` because of return type
   |                  |       help: consider using an impl return type: `impl Send`
   |                  this type parameter
```

My sense would be that "this type parameter" label could be replaced with the text above... Not sure how hard that is though.

Alternatively, "the type of `{}` is chosen by the caller" should probably be extended to explain its significance. Maybe something along the lines of "and may be a type that is different than {(the other type mentioned)}"...

Reply:
I updated the message to explain its significance. However finding exactly where "this type parameter" is derived from and put into the message here is a bit hard. Do let me know if this is sufficient, or if not some more context on how I can find the construction of the above "this type parameter"